docs/src/themes.tsx

```diff
@@ -98,20 +150,66 @@ export const appTheme = {
   dark: {
     colors: darkPalette,
   },
+  hcDark: {
+    colors: hcDarkPalette,
+  },
   media,
   breakpoints,
+  tmSelectors,
+};
+
+const themesArray = Object.values(ThemesEnum);
+
+export const getNextTheme = (currentTheme: ThemesEnum): ThemesEnum => {
+  const currentThemeIndex = themesArray.indexOf(currentTheme);
+  const nextThemeIndex =
+    currentThemeIndex === themesArray.length - 1 ? 0 : currentThemeIndex + 1;
+  const nextTheme = themesArray[nextThemeIndex];
+  return nextTheme;
 };
 
 const theming = createTheming(appTheme);
 
-type Theme = typeof appTheme;
+interface IThemeContext {
+  theme: ThemesEnum;
+  changeTheme: () => void;
+}
+
+export const ThemeContext = React.createContext<IThemeContext>({
+  theme: ThemesEnum.AUTO,
+  changeTheme: () => {},
+});
 
 export const ThemeProvider = ({
   children,
-  theme: themeProp,
-}: React.PropsWithChildren<{ theme: Theme }>): JSX.Element => (
-  <theming.ThemeProvider theme={themeProp}>{children}</theming.ThemeProvider>
-);
+  overrideTheme,
+}: React.PropsWithChildren<{ overrideTheme?: ThemesEnum }>): JSX.Element => {
+  const [theme, setTheme] = useState<ThemesEnum>(ThemesEnum.AUTO);
+
+  const changeTheme = useCallback(() => {
+    const body = document.querySelector("body") as Element;
+    const newTheme = overrideTheme || ThemesEnum[getNextTheme(theme)];
+    body.className = `${newTheme}`;
+    localStorage.setItem("theme", newTheme);
+    setTheme(newTheme);
+  }, [theme, setTheme, overrideTheme]);
+
+  const initialContext = useMemo(
+    () => ({ theme, changeTheme }),
+    [theme, changeTheme]
+  );
+
+  useEffect(() => {
+    const savedTheme = localStorage.getItem("theme") as ThemesEnum;
```

Comment:
Could you clarify can't we instead of useEffect do smth like this:

```js

const savedTheme = localStorage.getItem("theme") as ThemesEnum;
const [theme, setTheme] = useState<ThemesEnum>(savedTheme || ThemesEnum.AUTO);
```